### ISSUE TITLE
[a11y]: Accessible Name for RadioButtons

### ISSUE THREAD

**Issue by srikant-ch5** [carbon-design-system/carbon#17753]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.62.1

### React version

18.3.0

### Automated testing tool and ruleset

IBM Equal Accessibility Checker

### Description

In some scenarios we don't want to show legend text in UI for associated Radiobuttons in such case there is an accessibility violation `Group/Fieldset does not have an accessible name`.

### Reproduction/example

Run IBM Accessibility Checker on a RadioButtonGroup with no legendText.

### Steps to reproduce

Open IBM Accessibility Checker Tool to Scan RadioButtonGroup with no legendText.

![Image](https://github.com/user-attachments/assets/1685bb85-eea1-437f-9e04-731463aa8678)

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by nkn2022** [comment]
Adding `legendText` on the `RadioButtonGroup` had fixed this issue for me. You can try that. 
See example here: https://react.carbondesignsystem.com/?path=/docs/components-radiobutton--overview#overview

**Comment by guidari** [comment]
You can use the `aria-label` instead of the `legendText`. That way it will fix the a11y issue.

**Comment by gssspillai** [comment]
I am still experiencing an accessibility issue when passing the following props as arguments to the `Datagrid` of `@carbon/ibm-products`.

I tried adding `aria-labelledby` to resolve the accessibility issues. But the problem persists.

Any help would be greatly appreciated.

```

```
props: {
          RadioButtonGroup: {
            legendText: ‘Legend Text’,
            orientation: 'vertical',
            name: 'radio-button-group'
          },
          RadioButton: [
            {
              id: 'critical',
              labelText: ‘critical’,
              value: 'critical',
            },
            {
              id: 'warning',
              labelText: ‘critical’,
              value: 'warning',
            },
            {
              id: ‘criticalAndWarning’,
              labelText: ‘Critical and Warning’,
              value: ‘criticalAndWarning’,
            }
          ]
        }

```

```

**Comment by gssspillai** [comment]
Hi @guidari,
Could we please reopen this for further review? 
Thanks!
